be2-scala/src/main/scala/ch/epfl/pop/model/objects/ElectionChannel.scala

```diff
@@ -0,0 +1,64 @@
+package ch.epfl.pop.model.objects
+
+import akka.pattern.AskableActorRef
+import ch.epfl.pop.model.network.method.message.Message
+import ch.epfl.pop.model.network.method.message.data.election.{CastVoteElection, SetupElection}
+import ch.epfl.pop.pubsub.AskPatternConstants
+import ch.epfl.pop.storage.DbActor
+import ch.epfl.pop.storage.DbActor.{DbActorReadAck, DbActorReadChannelDataAck}
+
+import scala.collection.mutable
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{Await, Future}
+
+object ElectionChannel {
+  implicit class ElectionChannel(channel: Channel) extends AskPatternConstants {
+
+    /**
+     *
+     * @param dbActor AskableActorRef
+     * @tparam T type of data to extract from the election channel
+     * @return Future of a list of tuple containing the message and the data extracted
+     */
+    def extractMessages[T: Manifest](dbActor: AskableActorRef = DbActor.getInstance): Future[List[(Message, T)]] = {
```

Comment:
```suggestion
    def extractMessages[T <: MessageData: Manifest](dbActor: AskableActorRef = DbActor.getInstance): Future[List[(Message, T)]] = {
```

Optional, but I noticed the typing system inferred `T with MessageData` during the session so might as well.
(need to import `MessageData`)